3-enrich/stream-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/kinesis/sources/AbstractSource.scala

```diff
@@ -191,7 +191,16 @@ abstract class AbstractSource(config: KinesisEnrichConfig, igluResolver: Resolve
       validatedMaybeEvent match {
         case Success(co) => (tabSeparateEnrichedEvent(co), if (config.useIpAddressAsPartitionKey) {
             co.user_ipaddress
-          } else {
+          } 
+	  else if (config.partitionKey != null) {
+          try
+            String.valueOf(co.getClass.getDeclaredField(config.partitionKey).get(co))
```

Comment:
I'm not entirely sure it is going to work. As all JVM fields in Scala are private. Access to them is made only through getters.